src/compiler/factory.ts

```diff
@@ -24,14 +24,14 @@ namespace ts {
      * Make `elements` into a `NodeArray<T>`. If `elements` is `undefined`, returns an empty `NodeArray<T>`.
      */
     export function createNodeArray<T extends Node>(elements?: ReadonlyArray<T>, hasTrailingComma?: boolean): NodeArray<T> {
-        if (elements) {
+        if (!elements || elements === emptyArray) {
+            elements = [];

```

Comment:
This looks concerning. Does this happen often? Did you catch the code path where this occurres?

Reply:
I did look for exact code path since they are multiple ones but remember seeing this with code fix related node creations being one of them. it was discovered as part of #22167 when code fix related tests would cause those module resolution tests to fail....